### PR TITLE
Cleanup package (close #76)

### LINKED ISSUE #76: why `packageVersionIs` method exists?
I have not found usages of this `private` method
https://github.com/psalm/psalm-plugin-phpunit/blob/83b5c107a8902c6c7f0b4467685eac16b3f932b4/src/Plugin.php#L28-L38

May be it needs to be deleted?
Also if deleting this, also delete `composer/semver` dependency `composer remove composer/semver` because it's used only there

### PR DESCRIPTION
**Pull request by vladyslavstartsev**
close #76 
Actually I thought that I will only remove one method :smile: 

What I did is described in commits

### CODE REVIEW

File: src/Hooks/TestCaseHandler.php [hunk 1]
```diff
@@ -503,19 +500,16 @@ private static function getSpecials(ClassMethod $method): array
 
         if ($docblock) {
             try {
-                $parsed_comment = DocComment::parse(
-                    (string)$docblock->getReformattedText(),
-                    self::getCommentLine($docblock)
-                );
+                $parsed_comment = DocComment::parsePreservingLength($docblock);
```
Comment (weirdan) [inline src/Hooks/TestCaseHandler.php:503]: `DocComment::parsePreservingLength()` return type changed from shape to object at some point (@muglug, it seems we had an API break in a minor version). I suppose it's what causing low-deps build failures.

Also I suspect that `parsePreservingLength()` does not emit data on line numbers, which is needed later to properly highlight problematic lines in error snippets Psalm outputs with default reporter.

Comment (muglug) [reply src/Hooks/TestCaseHandler.php:503]: Crap, sorry, it was very stupid of me not to check this plugin. The method keys everything offsets, not line numbers

Comment (weirdan) [reply src/Hooks/TestCaseHandler.php:503]: @muglug It wasn't used before this pull request, so perhaps no one was affected. Might want to add https://github.com/Roave/BackwardCompatibilityCheck to CI pipeline though.

@vladyslavstartsev I'd suggest to revert this particular change then.

Comment (vladyslavstartsev) [reply src/Hooks/TestCaseHandler.php]: reverted @weirdan @muglug

**Comment by vladyslavstartsev** [comment]
@muglug Can I ask you for review?

**Comment by weirdan** [comment]
Think we can merge this without calling in the big guns.

**Pull request merged**